### ISSUE TITLE
taxon ranks in discoverlife variety and subspecies

### ISSUE THREAD

**Issue by seltmann** [globalbioticinteractions/nomer#57]
@jhpoelen there are two cases in the discoverlife taxonomy where the taxon rank should not be species

Tri-nomials (without a var) are considered **subspecies**. These are two examples of subspecies
https://www.discoverlife.org/mp/20q?search=Pseudopanurgus+nebrascensis+timberlakei	
Pseudopanurgus nebrascensis timberlakei

OR 

https://www.discoverlife.org/mp/20q?search=Protandrena+(Pterosarus)+labrosiformis+distractus	
Protandrena (Pterosarus) labrosiformis distractus

--------
Names (with a var) should be **variety** Here is an example.
https://www.discoverlife.org/mp/20q?search=Psithyrus+(Allopsithyrus)+barbutellus+var+bimaculatus	
Psithyrus (Allopsithyrus) barbutellus var bimaculatus

**Comment by jhpoelen** [comment]
@seltmann Thanks for your suggestions. I added the functionality, only to find out that all the pages linked from the main page appears to be species https://www.discoverlife.org/mp/20q?act=x_checklist&guide=Apoidea_species&flags=HAS . I am probably missing something. 

See attached output of 

```
$ nomer list discoverlife | gzip > discoverlife.tsv.gz 
```
Note that I haven't yet release a new version with latest features. 

[discoverlife.tsv.gz](https://github.com/globalbioticinteractions/nomer/files/7479579/discoverlife.tsv.gz)

**Comment by jhpoelen** [comment]
Also, note that:

```
$ cat discoverlife.tsv.gz | grep -v "species" | wc -l
0
```

Also, note that 

```
$ zcat discoverlife.tsv.gz | grep "Psithyrus.*bimaculatus"
https://www.discoverlife.org/mp/20q?search=Psithyrus+(Allopsithyrus)+barbutellus+var+bimaculatus	Psithyrus (Allopsithyrus) barbutellus var bimaculatus	HOMONYM_OF	no:match	no:match					
https://www.discoverlife.org/mp/20q?search=Psithyrus+(Metapsithyrus)+campestris+var+bimaculatus	Psithyrus (Metapsithyrus) campestris var bimaculatus	SYNONYM_OF	https://www.discoverlife.org/mp/20q?search=Bombus+campestris	Bombus campestris	species		Animalia | Arthropoda | Insecta | Hymenoptera | Apidae | Bombus campestris	https://www.discoverlife.org/mp/20q?search=Animalia | https://www.discoverlife.org/mp/20q?search=Arthropoda | https://www.discoverlife.org/mp/20q?search=Insecta | https://www.discoverlife.org/mp/20q?search=Hymenoptera | https://www.discoverlife.org/mp/20q?search=Apidae | https://www.discoverlife.org/mp/20q?search=Bombus+campestris	kingdom | phylum | class | order | family | species	https://www.discoverlife.org/mp/20q?search=Bombus+campestris	
https://www.discoverlife.org/mp/20q?search=Psithyrus+sylvestris+f+bimaculatus	Psithyrus sylvestris f bimaculatus	SYNONYM_OF	https://www.discoverlife.org/mp/20q?search=Bombus+sylvestris	Bombus sylvestris	species		Animalia | Arthropoda | Insecta | Hymenoptera | Apidae | Bombus sylvestris	https://www.discoverlife.org/mp/20q?search=Animalia | https://www.discoverlife.org/mp/20q?search=Arthropoda | https://www.discoverlife.org/mp/20q?search=Insecta | https://www.discoverlife.org/mp/20q?search=Hymenoptera | https://www.discoverlife.org/mp/20q?search=Apidae | https://www.discoverlife.org/mp/20q?search=Bombus+sylvestris	kingdom | phylum | class | order | family | species	https://www.discoverlife.org/mp/20q?search=Bombus+sylvestris	
```

suggesting that you example ```Psithyrus (Allopsithyrus) barbutellus var bimaculatus```  is only listed as a homonym, with unmatched name.

**Comment by seltmann** [comment]
@jhpoelen I thought the rank is for the primary name, not the valid name. You are right that all of the valid names in DiscoverLife are species names, but many of the synonyms are not. I could be wrong about the meaning of the TaxonRank column.

**Comment by jhpoelen** [comment]
Thanks for providing the context that only species are listed as valid names. The vars, subspecies etc. are being parsed, but they don't show up because  the left hand side of the ```(left name) -[:name_relation]-> (right name)``` only contains (name,id) , the right hand side has more information. 

Perhaps a reason to revisit the map format . . . and include more information on the left hand side (e.g., authorship name, rank, path etc.)

**Comment by jhpoelen** [comment]
Described functionality available in Nomer v0.2.8 . Please close this issue after review, or share review comments below.

**Comment by seltmann** [comment]
@jhpoelen looks good upon review. Closing issue.
